### ISSUE TITLE
Error in MatchRegionStats 1 not present in meta.features

### ISSUE THREAD

**Issue by ggeorgol** [timoast/signac#549]
Hello!

I am encountering an issue with the `FindMotifs` function which returns the error below:

```r
enriched.motifs <- FindMotifs(object = so_filt,features = te_peaks, assay = 'peaks', background = 20000)
```
>Error in MatchRegionStats(meta.feature = mf.choose, query.feature = mf.query, : 1 not present in meta.features
>Traceback:
>1. FindMotifs(object = so_filt, features = te_peaks, assay = "peaks", 
> .     background = 20000)
>2. MatchRegionStats(meta.feature = mf.choose, query.feature = mf.query, 
> .     regions = features, n = background, verbose = verbose, ...)
>3. stop(i, " not present in meta.features")

The same error arises with the `atac_small` data:

```r
test_peaks <- rownames(atac_small@assays$peaks@meta.features[1:100,])
FindMotifs(atac_small, features = test_peaks, background = 200)
```
>Selecting background regions to match input sequence characteristics
>
>Error in MatchRegionStats(meta.feature = mf.choose, query.feature = mf.query, : 1 not present in meta.features
>Traceback:
>
>1. FindMotifs(atac_small, features = test_peaks, background = 200)
>2. MatchRegionStats(meta.feature = mf.choose, query.feature = mf.query, 
>.     regions = features, n = background, verbose = verbose, ...)
>3. stop(i, " not present in meta.features")

Any help would be kindly appreciated.

Cheers!
--Grigorios

Here's what my `sessionInfo()` looks like

>R version 4.0.3 (2020-10-10)
>Platform: x86_64-conda-linux-gnu (64-bit)
>Running under: CentOS Linux 7 (Core)
>
>Matrix products: default
>BLAS/LAPACK: /vsc-hard-mounts/leuven-data/341/vsc34172/miniconda3/envs/r_env/lib/libopenblasp-r0.3.12.so
>
>locale:
>[1] C
>
>attached base packages:
>[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
>[8] methods   base     
>
>other attached packages:
> [1] eulerr_6.1.0              future.apply_1.7.0       
> [3] future_1.21.0             tidyr_1.1.3              
> [5] reticulate_1.18           leiden_0.3.7             
> [7] hdf5r_1.3.3               patchwork_1.1.1          
> [9] ggplot2_3.3.3             EnsDb.Hsapiens.v86_2.99.0
>[11] ensembldb_2.14.0          AnnotationFilter_1.14.0  
>[13] GenomicFeatures_1.42.2    AnnotationDbi_1.52.0     
>[15] Biobase_2.50.0            GenomicRanges_1.42.0     
>[17] GenomeInfoDb_1.26.4       IRanges_2.24.1           
>[19] S4Vectors_0.28.1          BiocGenerics_0.36.0      
>[21] SeuratObject_4.0.0        Seurat_4.0.1             
>[23] Signac_1.1.1             
>
>loaded via a namespace (and not attached):
>  [1] utf8_1.2.1                  tidyselect_1.1.0           
>  [3] RSQLite_2.2.5               htmlwidgets_1.5.3          
>  [5] grid_4.0.3                  BiocParallel_1.24.1        
>  [7] Rtsne_0.15                  munsell_0.5.0              
>  [9] codetools_0.2-18            ica_1.0-2                  
> [11] pbdZMQ_0.3-5                miniUI_0.1.1.1             
> [13] withr_2.4.1                 colorspace_2.0-0           
> [15] OrganismDbi_1.32.0          knitr_1.31                 
> [17] uuid_0.1-4                  rstudioapi_0.13            
> [19] ROCR_1.0-11                 tensor_1.5                 
> [21] listenv_0.8.0               labeling_0.4.2             
> [23] MatrixGenerics_1.2.1        repr_1.1.3                 
> [25] GenomeInfoDbData_1.2.4      polyclip_1.10-0            
> [27] farver_2.1.0                bit64_4.0.5                
> [29] parallelly_1.24.0           vctrs_0.3.7                
> [31] generics_0.1.0              xfun_0.22                  
> [33] biovizBase_1.38.0           BiocFileCache_1.14.0       
> [35] lsa_0.73.2                  ggseqlogo_0.1              
> [37] R6_2.5.0                    reshape_0.8.8              
> [39] bitops_1.0-6                spatstat.utils_2.1-0       
> [41] cachem_1.0.4                DelayedArray_0.16.3        
> [43] assertthat_0.2.1            promises_1.2.0.1           
> [45] scales_1.1.1                nnet_7.3-15                
> [47] gtable_0.3.0                globals_0.14.0             
> [49] goftest_1.2-2               ggbio_1.38.0               
> [51] rlang_0.4.10                RcppRoll_0.3.0             
> [53] splines_4.0.3               rtracklayer_1.50.0         
> [55] lazyeval_0.2.2              dichromat_2.0-0            
> [57] spatstat.geom_2.0-1         checkmate_2.0.0            
> [59] BiocManager_1.30.12         reshape2_1.4.4             
> [61] abind_1.4-5                 backports_1.2.1            
> [63] httpuv_1.5.5                SeuratDisk_0.0.0.9019      
> [65] Hmisc_4.5-0                 RBGL_1.66.0                
> [67] tools_4.0.3                 ellipsis_0.3.1             
> [69] spatstat.core_2.0-0         RColorBrewer_1.1-2         
> [71] ggridges_0.5.3              Rcpp_1.0.6                 
> [73] plyr_1.8.6                  base64enc_0.1-3            
> [75] progress_1.2.2              zlibbioc_1.36.0            
> [77] purrr_0.3.4                 RCurl_1.98-1.3             
> [79] prettyunits_1.1.1           rpart_4.1-15               
> [81] openssl_1.4.3               deldir_0.2-10              
> [83] pbapply_1.4-3               cowplot_1.1.1              
> [85] zoo_1.8-9                   SummarizedExperiment_1.20.0
> [87] ggrepel_0.9.1               cluster_2.1.1              
> [89] magrittr_2.0.1              RSpectra_0.16-0            
> [91] data.table_1.14.0           scattermore_0.7            
> [93] lmtest_0.9-38               RANN_2.6.1                 
> [95] SnowballC_0.7.0             ProtGenerics_1.22.0        
> [97] fitdistrplus_1.1-3          matrixStats_0.58.0         
> [99] hms_1.0.0                   mime_0.10                  
>[101] evaluate_0.14               xtable_1.8-4               
>[103] XML_3.99-0.6                jpeg_0.1-8.1               
>[105] gridExtra_2.3               compiler_4.0.3             
>[107] biomaRt_2.46.3              tibble_3.1.0               
>[109] KernSmooth_2.23-18          crayon_1.4.1               
>[111] htmltools_0.5.1.1           mgcv_1.8-34                
>[113] later_1.1.0.1               Formula_1.2-4              
>[115] DBI_1.1.1                   tweenr_1.0.2               
>[117] dbplyr_2.1.0                MASS_7.3-53.1              
>[119] rappdirs_0.3.3              Matrix_1.3-2               
>[121] cli_2.3.1                   igraph_1.2.6               
>[123] pkgconfig_2.0.3             GenomicAlignments_1.26.0   
>[125] foreign_0.8-81              IRdisplay_1.0              
>[127] plotly_4.9.3                spatstat.sparse_2.0-0      
>[129] xml2_1.3.2                  XVector_0.30.0             
>[131] stringr_1.4.0               VariantAnnotation_1.36.0   
>[133] digest_0.6.27               sctransform_0.3.2          
>[135] RcppAnnoy_0.0.18            graph_1.68.0               
>[137] spatstat.data_2.1-0         Biostrings_2.58.0          
>[139] polylabelr_0.2.0            fastmatch_1.1-0            
>[141] htmlTable_2.1.0             uwot_0.1.10                
>[143] curl_4.3                    shiny_1.6.0                
>[145] Rsamtools_2.6.0             lifecycle_1.0.0            
>[147] nlme_3.1-152                jsonlite_1.7.2             
>[149] viridisLite_0.3.0           askpass_1.1                
>[151] BSgenome_1.58.0             fansi_0.4.2                
>[153] pillar_1.5.1                GGally_2.1.1               
>[155] lattice_0.20-41             fastmap_1.1.0              
>[157] httr_1.4.2                  survival_3.2-10            
>[159] glue_1.4.2                  png_0.1-7                  
>[161] bit_4.0.4                   ggforce_0.3.3              
>[163] stringi_1.5.3               blob_1.2.1                 
>[165] latticeExtra_0.6-29         memoise_2.0.0              
>[167] IRkernel_1.1.1              dplyr_1.0.5                
>[169] irlba_2.3.3

**Comment by timoast** [comment]
Hi @ggeorgol, thanks for the bug report. This was mainly an issue with the error message, it should tell you that the GC content has not been computed and you need to run `RegionStats()` on the assay first. This has now been fixed on the develop branch.

**Comment by davidcoffey** [comment]
It appears this bug his not been resolved in the latest version of signac v1.10.0.  Below is a reproducible example:

```r
DefaultAssay(atac_small) = "peaks"

FindMotifs(atac_small, features = row.names(atac_small))`
```
> Selecting background regions to match input sequence characteristics
> Error in MatchRegionStats(meta.feature = mf.choose, query.feature = mf.query,  : 
>   meta.feature should be a data.frame

I made sure the region stats have been computed as recommended by @timoast 
```r
atac_small = RegionStats(atac_small, genome = BSgenome.Hsapiens.UCSC.hg38)`
```

I then confirmed the region stats are represented in the meta.features slot
```r
head(atac_small@assays$peaks@meta.features)
```
>  count percentile  AA  AC  AG  AT  CA CC CG  CT  GA GC  GG GT  TA  TC  TG  TT GC.percent sequence.length
> chr1-713460-714823    81  0.8142415 127  77  54 162 100 55  5 107  57 30  31 64 137 104  92 161   32.91789            1364
> chr1-752422-753038    16  0.5325077  44  22  52  41  44 43  3  59  40 32  40 25  31  52  42  46   46.51540             617
> chr1-762106-763359    47  0.7337461 174  49 149  73  83 43  1  69 116 55 127 44  72  49  64  85   42.90271            1254
> chr1-779589-780271     9  0.3962848  53  33  48  43  47 31 20  48  45 46  46 31  32  35  54  70   45.97365             683
> chr1-804872-805761    24  0.6191950 115  60  56  63  84 49  8  62  38 38  32 40  58  56  52  78   39.43820             890
> chr1-839520-841123    28  0.6656347 222 117 115 133 152 60 15  80 102 55  63 75 111  74 102 127   37.53117            1604

I next confirmed that the meta.features object is a data.frame as the error message eludes to
```r
class(atac_small@assays$peaks@meta.features)
```
> [1] "data.frame"

However, I identifed that the MatchRegionStats function pulls the meta.features using the `GetAssayData` function that seems to extract the counts as a dgCMatrix rather than meta.features data.frame
```r
class(GetAssayData(object = atac_small, assay = "peaks", slot = "meta.features"))`
```
> [1] "dgCMatrix"
> attr(,"package")
> [1] "Matrix"

Therefore, it seems the underlying issue is with the `GetAssayData`.  Interestingly enough, I cannot reproduce this issue using the same version of Signac on our linux server running the same version of R.

```r
sessionInfo()
```

> R version 4.2.3 (2023-03-15)
> Platform: x86_64-apple-darwin17.0 (64-bit)
> Running under: macOS Ventura 13.5
> 
> Matrix products: default
> LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
> 
> locale:
> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
> 
> attached base packages:
> [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
>  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.66.3                   rtracklayer_1.58.0                Biostrings_2.66.0                 XVector_0.38.0                    GenomicRanges_1.50.2             
>  [7] GenomeInfoDb_1.34.9               IRanges_2.32.0                    S4Vectors_0.36.2                  BiocGenerics_0.44.0               Signac_1.10.0                     Seurat_4.9.9.9059                
> [13] SeuratObject_4.9.9.9091           sp_1.6-0                         
> 
> loaded via a namespace (and not attached):
>   [1] spam_2.9-1                  fastmatch_1.1-3             BiocFileCache_2.6.1         plyr_1.8.8                  igraph_1.4.2                lazyeval_0.2.2              splines_4.2.3               RcppHNSW_0.4.1             
>   [9] BiocParallel_1.32.6         listenv_0.9.0               scattermore_0.8             ggplot2_3.4.2               digest_0.6.31               htmltools_0.5.5             fansi_1.0.4                 magrittr_2.0.3             
>  [17] memoise_2.0.1               tensor_1.5                  cluster_2.1.4               ROCR_1.0-11                 globals_0.16.2              matrixStats_0.63.0          spatstat.sparse_3.0-1       prettyunits_1.1.1          
>  [25] colorspace_2.1-0            blob_1.2.4                  rappdirs_0.3.3              ggrepel_0.9.3               dplyr_1.1.2                 crayon_1.5.2                RCurl_1.98-1.12             jsonlite_1.8.4             
>  [33] progressr_0.13.0            spatstat.data_3.0-1         survival_3.5-3              zoo_1.8-12                  glue_1.6.2                  polyclip_1.10-4             gtable_0.3.3                zlibbioc_1.44.0            
>  [41] leiden_0.4.3                DelayedArray_0.24.0         future.apply_1.10.0         abind_1.4-5                 scales_1.2.1                DBI_1.1.3                   spatstat.random_3.1-4       miniUI_0.1.1.1             
>  [49] Rcpp_1.0.10                 viridisLite_0.4.1           xtable_1.8-4                progress_1.2.2              reticulate_1.28             bit_4.0.5                   dotCall64_1.0-2             htmlwidgets_1.6.2          
>  [57] httr_1.4.5                  RColorBrewer_1.1-3          ellipsis_0.3.2              ica_1.0-3                   pkgconfig_2.0.3             XML_3.99-0.14               dbplyr_2.3.2                uwot_0.1.14                
>  [65] deldir_1.0-6                utf8_1.2.3                  tidyselect_1.2.0            rlang_1.1.0                 reshape2_1.4.4              later_1.3.0                 AnnotationDbi_1.60.2        munsell_0.5.0              
>  [73] tools_4.2.3                 cachem_1.0.7                cli_3.6.1                   generics_0.1.3              RSQLite_2.3.1               ggridges_0.5.4              stringr_1.5.0               fastmap_1.1.1              
>  [81] yaml_2.3.7                  goftest_1.2-3               bit64_4.0.5                 fitdistrplus_1.1-11         purrr_1.0.1                 RANN_2.6.1                  SeuratData_0.2.2.9001       KEGGREST_1.38.0            
>  [89] pbapply_1.7-0               future_1.32.0               nlme_3.1-162                mime_0.12                   RcppRoll_0.3.0              xml2_1.3.3                  biomaRt_2.54.1              compiler_4.2.3             
>  [97] rstudioapi_0.14             filelock_1.0.2              curl_5.0.0                  plotly_4.10.1               png_0.1-8                   spatstat.utils_3.0-2        tibble_3.2.1                stringi_1.7.12             
> [105] GenomicFeatures_1.50.4      RSpectra_0.16-1             lattice_0.20-45             Matrix_1.5-3                vctrs_0.6.2                 pillar_1.9.0                lifecycle_1.0.3             spatstat.geom_3.1-0        
> [113] lmtest_0.9-40               RcppAnnoy_0.0.20            data.table_1.14.8           cowplot_1.1.1               bitops_1.0-7                irlba_2.3.5.1               httpuv_1.6.9                patchwork_1.1.2            
> [121] BiocIO_1.8.0                R6_2.5.1                    promises_1.2.0.1            KernSmooth_2.23-20          gridExtra_2.3               parallelly_1.35.0           codetools_0.2-19            fastDummies_1.6.3          
> [129] MASS_7.3-58.2               SummarizedExperiment_1.28.0 rjson_0.2.21                GenomicAlignments_1.34.1    Rsamtools_2.14.0            sctransform_0.3.5           GenomeInfoDbData_1.2.9      parallel_4.2.3             
> [137] hms_1.1.3                   grid_4.2.3                  tidyr_1.3.0                 MatrixGenerics_1.10.0       Rtsne_0.16                  spatstat.explore_3.1-0      Biobase_2.58.0              shiny_1.7.4                
> [145] restfulr_0.0.15

**Comment by timoast** [comment]
@davidcoffey you need to install from the seurat5 branch if you have installed Seurat v5:

```r
remotes::install_github("stuart-lab/signac", "seurat5", quiet = TRUE)
```

Seurat v5 changed how the meta features are accessed which is why it causes an error with the current Signac release.
